include/bringauto/fleet_protocol/http_client/FleetApiClient.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <CppRestOpenAPIClient/api/CarApi.h>
+#include <CppRestOpenAPIClient/api/DeviceApi.h>
+#include <CppRestOpenAPIClient/api/ModuleApi.h>
+#include <bringauto/fleet_protocol/http_client/RequestFrequencyGuard.hpp>
+
+#include <optional>
+
+
+namespace bringauto::fleet_protocol::http_client {
+
+
+class FleetApiClient {
+public:
+	/**
+     * @brief Constructs the fleet API client
+     * @param apiUrl URL of Fleet v2 HTTP API
+     * @param apiKey api key used for authorization
+     * @param companyName company identifier in the API
+     * @param carName car identifier in the API
+     * @param maxRequestsThresholdCount max amount of allowed requests in the specified time period (only for getCommands/getStatuses)
+     * @param maxRequestsThresholdPeriodMs time period in which the amount of requests is checked (only for getCommands/getStatuses)
+     * @param delayAfterThresholdReachedMs time to sleep for in ms when threshold is initially reached (only for getCommands/getStatuses)
+     * @param retryRequestsDelayMs delay in ms between requests until request rate is no longer over the threshold (only for getCommands/getStatuses)
+     */
+	FleetApiClient(const std::string &apiUrl, const std::string &apiKey, const std::string &companyName, const std::string &carName,
+				   int32_t maxRequestsThresholdCount = 10, int32_t maxRequestsThresholdPeriodMs = 5000,
+				   int32_t delayAfterThresholdReachedMs = 5000, int32_t retryRequestsDelayMs = 200);
```

Comment:
Maybe encapsulate the params into two structures? default values can be set in that structure. The code will be more readable

Reply:
extracted the params into structs